libafl/src/corpus/inmemory_ondisk.rs

```diff
@@ -229,13 +230,27 @@ where
         Self::_new(dir_path.as_ref(), None)
     }
 
+    ///Create an [`InMemoryOnDiskCorpus`] that will not store .`lafl_lock` files
```

Comment:
The docs should clearly state that this will be racey if multiple cores work on the same corpus.

Maybe I'd even rename this function `multicore_unsafe`or similar. The fact that lafl_lock files are produced is just an impelmentation detail imho, what do you think?

Reply:
I think the .lafl_lock file is needed if working on multi-core. In some cases, if there is no competition problem for a single core, the user is allowed not to generate the .lafl_lock file, so I did not consider the competition problem without generating the .lafl_lock file.